src/algebra/ordered_ring.lean

```diff
@@ -670,6 +672,13 @@ lemma mul_self_lt_mul_self_iff {a b : α} (h1 : 0 ≤ a) (h2 : 0 ≤ b) : a < b
 iff.trans (lt_iff_not_ge _ _) $ iff.trans (not_iff_not_of_iff $ mul_self_le_mul_self_iff h2 h1) $
   iff.symm (lt_iff_not_ge _ _)
 
+lemma eq_if_mul_self_eq_mul_self {a b : α} (h1 : 0 ≤ a) (h2 : 0 ≤ b) (h3 : a * a = b * b) : a = b :=
```

Comment:
`eq_if` -> `eq_of`. Better yet, this should be an iff on the last hypothesis, and then it can be called `mul_self_inj`.

Reply:
Done, thanks!